draft/XXXX-upload-system/README.md

```diff
@@ -0,0 +1,126 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Status: draft
+Created: 2020-07-08
+Last-Modified: 2020-07-08
+---
+
+# RFC XXXX: Advanced and Robust Upload System
+
+Uploading a large dataset can be both timeconsuming and problematic. This RFC proposes a method for handling uploads that both facilitates ease of use, as well as reliability and fault tolerance. Uploading data is one of the most important parts of this system, and as such should be robust and featured.
+
+## Motivation
+
+Our current system of dealing with uploads is very simple and unfeatured. We send the data through a single request, with no track of progress or anything. This exposes a couple of key issues:
+
+* If the upload takes too long, the connection times out (due to a current gunicorn setting)
+* If the connection is interrupted, the upload must start over from the beginning (very frustrating on large datasets)
+* There's no progress/indication to the user. This can be implemented on the _client_ by simply tracking the HTTP request progress, but as a matter of exposing this functionality through our API, it does not exist.
+
+## Proposals
+
+### Chunked Uploads
+Instead of dealing with a single request to upload data, this data could instead be sent in chunks, allowing for many smaller HTTP requests. This would allow the following things:
+* Retain a timeout on requests, while also allowing large datasets to be uploaded with ease.
```

Comment:
There are multiple timeouts for requests to consider. Often times, for large file uploads or downloads, you want  to leave the connection open as long as it takes, and the timeout you care about is the timeout between data packets being sent. (If you're worried about security, the thing to do is to validate the content length before reading the body, and if the body goes over it, you kill the connection and abort the upload.)